File: app/src/main/java/intellij/class6/herancas/exercise/java/Conta.java
package intellij.class6.herancas.exercise.java;

public abstract class Conta {
    private double saldo = 0.0;
    private Cliente titular;

    public Conta() {
    }

    public Conta(Cliente titular) {
        setSaldo(saldo);
        setTitular(titular);
    }

    public Conta(double saldo, Cliente titular) {
        setSaldo(saldo);
        setTitular(titular);
    }

    public double cashDeposit(double value) {
        setSaldo(consultarSaldo() + value);
        System.out.println("Depósito Dinheiro. Saldo: " + consultarSaldo());
        return consultarSaldo();
    }

    public double withdrawMoney(double value) {
        if (value <= consultarSaldo()) {
            setSaldo(consultarSaldo() - value);
            System.out.println("Saque. Saldo: " + consultarSaldo());
            return consultarSaldo();
        }
        System.out.println("Saldo Insuficiente.");
        return 0.0;
    }

    public double consultarSaldo() {
        return saldo;
    }

    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

    public Cliente getTitular() {
        return titular;
    }

    public void setTitular(Cliente titular) {
        this.titular = titular;
    }
}

File: src/exercise03/kotlin/Conta.kt
package exercise03.kotlin

abstract class Conta {
    private var saldo = 0.0
    var titular: Cliente? = null

    constructor() {}
    constructor(titular: Cliente) {
        setSaldo(saldo)
        this.titular = titular
    }

    constructor(saldo: Double, titular: Cliente) {
        setSaldo(saldo)
        this.titular = titular
    }

    fun cashDeposit(value: Double): Double {
        setSaldo(consultarSaldo() + value)
        println("Depósito Dinheiro. Saldo: " + consultarSaldo())
        return consultarSaldo()
    }

    open fun withdrawMoney(value: Double): Double {
        if (value <= consultarSaldo()) {
            setSaldo(consultarSaldo() - value)
            println("Saque. Saldo: " + consultarSaldo())
            return consultarSaldo()
        }
        println("Saldo Insuficiente.")
        return 0.0
    }

    fun consultarSaldo(): Double {
        return saldo
    }

    fun setSaldo(saldo: Double) {
        this.saldo = saldo
    }

}